src/main/fc/rc.c

```diff
@@ -489,141 +492,176 @@ static FAST_CODE void processRcSmoothingFilter(void)
-            // don't add acceleration, empirically seems better on FrSky
-        }
-        setpointSpeed *= rxRate;
-        prevDuplicatePacket[axis] = false;
-    } else {
-        // no movement!
-        if (prevDuplicatePacket[axis] == false) {
-            // first duplicate after movement
-            setpointSpeed = prevSetpointSpeed[axis];
-            if (fabsf(setpoint) < 0.95f * maxRcRate[axis]) {
-                setpointSpeed += prevAcceleration[axis];
+    float setpointSpeed = 0.0f;
+    float setpointSpeedDelta = 0.0f;
+    float feedforward = 0.0f;
+
+    if (pid->feedforwardInterpolate) {
+        static float prevRxInterval;
+        // for Rx links which do send duplicate data packets when they send telemetry
+        // interpolate setpointSpeed to minimise steps in feedforward
+        if (rcCommandDeltaAbs) {
+            // movement!
+            // if the previous packet was a duplicate, calculate setpointSpeed over the last two intervals
+            if (prevPacketWasADuplicate[axis]) {
+                rxRate = 1.0f / (rxInterval + prevRxInterval);
             }
-            zeroTheAcceleration = 0.0f; // force acceleration to zero
+            setpointSpeed = setpointDelta * rxRate;
+            prevPacketWasADuplicate[axis] = false;
         } else {
-            // second and subsequent duplicates after movement should be zeroed
-            setpointSpeed = 0.0f;
-            prevSetpointSpeed[axis] = 0.0f;
-            zeroTheAcceleration = 0.0f; // force acceleration to zero
+            // no movement!
+            if (!prevPacketWasADuplicate[axis]) {
+                // interpolate a replacement setpointSpeed value for the first duplicate after normal movement
+                // avoid interpolating when about to hit max deflection
+                if (fabsf(setpoint) < 0.90f * maxRcRate[axis]) {
+                    // we assume here that this was a single packet duplicate, and that it is of approximately normal duration
+                    // hence no modification of prevSetpointSpeedDelta on the basis of rxInterval/preRxInterval
+                    // we could do add prevSetpointSpeedDelta[axis] * rxInterval / preRxInterval
+                    // but I think this won't materially improve the extrapolation
+                    setpointSpeed = prevSetpointSpeed[axis] + prevSetpointSpeedDelta[axis];
+                    // pretend that there was stick movement also, to hold the same jitter value
+                    rcCommandDeltaAbs = prevRcCommandDeltaAbs[axis];
+                }
+            } else {
+                // for second and all subsequent duplicates...
+                // leave setpoint speed at zero
+                // zero the acceleration by setting previous speed to zero
+                // feedforward will smoothly decay and be attenuated by the jitter reduction value for zero rcCommandDelta
+                prevSetpointSpeed[axis] = 0.0f; // zero acceleration later on
```

Comment:
Maybe be explicit (it should be optimized away anyway)
```suggestion
                setpointSpeed = 0.0f;
                prevSetpointSpeed[axis] = 0.0f; // zero acceleration later on
```

Reply:
If we get here, `setpointSpeed` cannot have been modified from where it is set to zero with `float setpointSpeed = 0.0f;`.
I agree that it is more clear if we are explicit, and it doesn't make the code bigger, so it appears to be compiled away.
Thanks.

Reply:
Yes, you version is correct. But when reading code, you need to look back to check bash to see default value. So this is slightly easier to read